### PR TITLE
jsonrpc: replace static Addon.Types type with dynamically created one to fix missing addon types

### PR DESCRIPTION
**Pull request by Montellese**
Mizaki reported a bug in the forum that JSON-RPC's `Addons.GetAddons` returns addons with types that aren't part of the `Addon.Types` enumeration defined in the API definition. This is obviously because we forget to update the API definition when we add new addon types.

To fix that I have changed `Addon.Types` from a statically defined to a dynamically generated enum type which is automatically filled in during the initialization of the JSON-RPC API definition. We also do this for other types like `Input.Action` and `GUI.Window`.

### CODE REVIEW

File: xbmc/interfaces/json-rpc/JSONRPC.cpp [hunk 1]
```diff
@@ -45,6 +47,10 @@ void CJSONRPC::Initialize()
 
   // Add some types/enums at runtime
   vector<string> enumList;
+  for (int addonType = ADDON::ADDON_UNKNOWN; addonType < ADDON::ADDON_MAX; addonType++)
+    enumList.push_back(ADDON::TranslateType(static_cast<ADDON::TYPE>(addonType), false));
+  CJSONServiceDescription::AddEnum("Addon.Types", enumList);
+
```
Comment (da-anda) [inline xbmc/interfaces/json-rpc/JSONRPC.cpp]: I think you're missing a enumList.clear(); here

Comment (Montellese) [inline xbmc/interfaces/json-rpc/JSONRPC.cpp]: Nice spotting, thanks.

**Comment by Montellese** [comment]
Fixed the issue noticed by @da-anda. jenkins build this please

**Comment by Montellese** [comment]
Bumped the API's bugfix version number. jenkins build and merge

**Pull request merged**